progect-management-app/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -0,0 +1,238 @@
+      .subscribe(
+        (columns: Column[]) => {
+          if (columns.length > 0) {
+            this.getColumn(this.boardId)
+          } else {
+            this.openModuleForm()
+          }
+        })
+  }
+ openModuleForm(){
+   const dialogRef = this.dialog.open(ColumnFormComponent);
+   dialogRef.componentInstance.columnCreated.subscribe((formData: any) => {
+     if(formData){
+       const columnData: Column = {
+         _id: this.boardId,
+         title: formData.title,
+         order: formData.order
+       };
+       this.createColumn(columnData);
+       dialogRef.close();
+     }
+   });
+ }
+  createColumn(columnData: Column) {
+    this.dashboardsService.createColumn(columnData.title, columnData.order, this.boardId)
+      .subscribe(
+        (res) => {
+          console.log(res);
+          this.getColumn(this.boardId);
+      })
+  }
+
+  getColumn(boardColumnId: string) {
+    this.dashboardsService.getColumn(boardColumnId)
+      .subscribe(
+        (columns: Column[]) => {
+          this.columns = columns.map(
+            column=> {
+            return { ...column, ...{tasks: []} };
+          })
+          this.columns.forEach(column => {
+            this.getTasks(column.boardId, column._id);
+          });
+        })
+  }
+
+  deleteColumn(boardId: string, columnId: string) {
+    const dialogRef = this.dialog.open(DialogComponent,{
+      data: {
+         message: 'Are you sure you want to delete this column?'
+       }
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result === true) {
+        this.dashboardsService.deleteColumn(boardId, columnId)
+          .subscribe(
+            (res) => {
+              this.getColumn(this.boardId);
+            }
+          );
+      }
+    })
+  }
+  goBack(){
+    this.location.back();
+  }
+
+  createTasks(boardId: string, columnId: string,taskData: Task){
+    this.dashboardsService.addTask(
+      boardId, columnId, taskData.title, taskData.order!,taskData.description,taskData.userId as number, taskData.users!
+      )
+      .subscribe(
+        (task: Task) => {
+          this.getTasks(boardId, columnId)
+        })
+  }
+
+  getTasks(boardId: string, columnId: string) {
+    this.dashboardsService.getTasks(boardId, columnId)
+      .pipe(
+        map((tasks: Task[]) => {
+          this.columns = this.columns.map(column => {
+            if (column._id === columnId) {
+              column.tasks = tasks;
+            }
+            return column;
+          });
+          return tasks;
+        }),
+        first()
+      )
+      .subscribe();
+  }
+  deleteTask(boardId:string,columnId: string, taskId:string){
+    const dialogRef = this.dialog.open(DialogComponent,{
+      data: {
+        message: 'Are you sure you want to delete this task?'
+      }
+    });
+    dialogRef.afterClosed().subscribe(result => {
+      if (result === true) {
+        this.dashboardsService.deleteTask(boardId, columnId, taskId)
+          .subscribe(
```

Comment:
Subscription inside another subscription – kind of bad practice. I would recommend you to look deeper in the RX pipes like switchMap